app/controllers/comments_controller.rb

```diff
@@ -1,23 +1,19 @@
 class CommentsController < ApplicationController
-  before_action :set_post, only: [:create]
-
   def create
+    @post = Post.find(params[:post_id])
     @comment = @post.comments.build(comment_params)
     @comment.user = current_user
-
-    if @comment.save
-      redirect_to post_path(@post), notice: 'Comment was successfully created.'
-    else
-      redirect_to post_path(@post), alert: 'Error creating comment.'
-    end
+    @comment.save
+    redirect_to post_path(@post)
   end
 
-  private
-
-  def set_post
+  def new
```

Comment:
- [ ] In your config/routes.rb file, you should have a route for new comments.
`GET   /posts/:post_id/comments/new(.:format)     comments#new`